src/storage/UpdateVertexProcessor.cpp

```diff
@@ -341,12 +341,25 @@ void UpdateVertexProcessor::process(const cpp2::UpdateVertexRequest& req) {
             return std::string("");
         },
         [this, partId, vId, req] (kvstore::ResultCode code) {
-            this->pushResultCode(this->to(code), partId);
-            if (code == kvstore::ResultCode::SUCCEEDED) {
-                onProcessFinished(req.get_return_columns().size());
-            } else {
-                LOG(ERROR) << "Failure update vertex, spaceId: " << this->spaceId_
+            while (true) {
+                if (code == kvstore::ResultCode::SUCCEEDED) {
+                    onProcessFinished(req.get_return_columns().size());
+                    break;
+                }
+                LOG(ERROR) << "Fail to update vertex, spaceId: " << this->spaceId_
                            << ", partId: " << partId << ", vId: " << vId;
+                if (code == kvstore::ResultCode::ERR_LEADER_CHANGED) {
+                    auto addrRet = kvstore_->partLeader(this->spaceId_, partId);
+                    if (ok(addrRet)) {
+                        auto leader = value(std::move(addrRet));
+                        this->pushResultCode(to(code), partId, leader);
+                        break;
+                    }
+                    LOG(ERROR) << "Fail to get part leader, spaceId: " << this->spaceId_
+                               << ", partId: " << partId << ", ResultCode: " << error(addrRet);
+                }
+                this->pushResultCode(to(code), partId);
```

Comment:
Why not extract these same logics to a function like `handleLeaderChanged`?

Reply:
> Why not extract these same logics to a function like `handleLeaderChanged` in base class?

No reason to refuse you...